addons/udes_stock_refactoring/tests/test_splitting.py

```diff
@@ -0,0 +1,649 @@
+        self._assert_move_fields(apple_pick_move, self.apple, 10)
+
+        apple_pick_move_line = apple_pick.move_line_ids
+        self._assert_move_line_fields(apple_pick_move_line, self.apple, 10, apple_pallet)
+
+        other_apple_pick_move = other_apple_pick.move_lines
+        self._assert_move_fields(other_apple_pick_move, self.apple, 10)
+
+        other_apple_pick_move_line = other_apple_pick.move_line_ids
+        self._assert_move_line_fields(other_apple_pick_move_line, self.apple, 10, other_apple_pallet)
+
+        # Check that the original pick has been reused
+        self.assertTrue(self.picking.exists())
+
+    def test_reserve_1_pallet_diff_prods_splits_1_pick(self):
+        """
+        With 1 pallet that contains different products.
+        Reserve pick (action_assign).
+        Original pick should be maintained.
+        """
+        fig = self.fig
+        grape = self.grape
+
+        mixed_pallet = self.create_package()
+
+        self.create_quant(fig.id, self.test_stock_location_01.id, 5, package_id=mixed_pallet.id)
+        self.create_quant(grape.id, self.test_stock_location_01.id, 10, package_id=mixed_pallet.id)
+
+        products_info = [{"product": fig, "qty": 5}, {"product": grape, "qty": 10}]
+        moves = self.create_move(self.picking, products_info)
+
+        fig_move = moves.filtered(lambda m: m.product_id == fig)
+        grape_move = moves - fig_move
+
+        self.picking.action_assign()
+
+        pick = (fig_move | grape_move).picking_id
+
+        self._assert_picking_fields(pick, state="assigned", group_name=mixed_pallet.name)
+
+        # Check we haven't mangled the moves or move_lines
+        moves = pick.move_lines
+        self.assertEqual(len(moves), 2)
+
+        fig_move = moves.filtered(lambda m: m.product_id == fig)
+        grape_move = moves - fig_move
+
+        self.assertEqual(fig_move.product_uom_qty, 5)
+        self.assertEqual(grape_move.product_uom_qty, 10)
+
+        move_lines = pick.move_line_ids
+        self.assertEqual(len(move_lines), 2)
+
+        fig_moveline = move_lines.filtered(lambda ml: ml.product_id == fig)
+        grape_moveline = move_lines - fig_moveline
+
+        self._assert_move_line_fields(fig_moveline, fig, 5, mixed_pallet)
+        self._assert_move_line_fields(grape_moveline, grape, 10, mixed_pallet)
+
+        # Check that the original pick has been reused
+        self.assertTrue(self.picking.exists())
+
+    def test_reserve_simultaneously_2_picks_1_pallet_same_prods_splits_1_pick(self):
+        """
+        With 2 picks for 2 of the same products on one pallet.
+        Reserve both picks (action_assign) simultaneously.
+        Original pick should be marked as empty and deleted.
+        A new pick should be created with both the moves.
+        """
+        apple_pallet = self.apple_pallet
+
+        self.create_quant(self.apple.id, self.test_stock_location_01.id, 15, package_id=apple_pallet.id)
```

Comment:
No need to create a new quant, just increase the quantity to 25.

Reply:
Have now done this.